adding back in the generate jwk using api step (#3251)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/implement-oauth-for-okta-serviceapp/main/index.md

## Create a service app integration

Create an OAuth 2.0 service app integration using the Admin Console.

  > **Note:** You can also use the `/oauth2/v1/clients` endpoint to [create your service app using the API](/docs/reference/api/oauth-clients/#request-example-create-a-service-app-with-a-jwks). If you use the API, follow the [Generate the JWK using the API](#generate-the-jwk-using-the-api) section below first, as you need the `JWKS` parameter value when you create the client using the API. 
  >
  >You can also add additional JWKS to the app later using the [Add a new JSON Web Key](https://developer.okta.com/docs/reference/api/apps/#add-new-json-web-key) API.

1. Sign in to your Okta organization as a user with administrative privileges. [Create an org for free](https://developer.okta.com/signup).

2. In the Admin Console, go to **Applications** > **Applications**, and then click **Create App Integration**.

3. On the Create a new app integration page, select **API Services** as the **Sign-in method** and click **Next**.

4. Enter a name for your app integration and click **Save**.

> ## Generate the JWK using the API

Reviewer: This is a question for my own knowledge: when do we use JWK vs JWKS?

> The `private_key_jwt` client authentication method is the only supported method for OAuth service apps that want to get access tokens with Okta scopes.

Reviewer: ```suggestion
The `private_key_jwt` client authentication method is the only supported method for OAuth 2.0 service apps that want to get access tokens with Okta scopes.
```

> The private key that you use to sign the JWT must have the corresponding public key registered in the [JWKSet](/docs/reference/api/oauth-clients/#json-web-key-set) of the OAuth service app. We recommend generating the public/private key pair first before creating the OAuth service app.

Reviewer: ```suggestion
The private key that you use to sign the JWT must have the corresponding public key registered in the [JWKSet](/docs/reference/api/oauth-clients/#json-web-key-set) of the OAuth 2.0 service app. Generate the public/private key pair first before creating the OAuth 2.0 service app.
```

1. Use a tool such as this [JSON Web Key Generator](https://mkjwk.org/) to generate a JWKS public/private key pair for testing. Okta supports both RSA and Elliptic Curve (EC) keys. In this example, we are selecting **RSA** as the encryption algorithm. Select the following values:

    * Key size: 2048
    * Key use: signature
    * Algorithm: RSA256
    * Key ID: This can be any random value.

> **Note:** Use the JSON Web Key Generator link to generate a JWKS public/private key pair for testing purposes only. For a production use case, use your own internal instance of the key pair generator. See this [key pair generator](https://github.com/mitreid-connect/mkjwk.org) for an example.

2. The JSON Web Key Generator tool extracts the public key from the key pair automatically. For testing purposes, copy the Public Key that is provided.

> > **Note:** Some Okta SDKs require that keys be in Privacy Enhanced Mail (PEM) format. If you are working with an Okta SDK that requires that the key be in PEM format, after you have generated the key pair, copy the public/private key pair into a [JWK to PEM Convertor tool](https://8gwifi.org/jwkconvertfunctions.jsp) and copy the private key to use when signing the JWT.

Reviewer: ```suggestion
> **Note:** Some Okta SDKs require that keys be in Privacy Enhanced Mail (PEM) format. If you are working with an Okta SDK that requires that the key be in PEM format, after you have generated the key pair, copy the public/private key pair into a [JWK to PEM Convertor tool](https://8gwifi.org/jwkconvertfunctions.jsp) and use the PEM formatted private key when signing the JWT.
```
